Mail2Bug/WorkItemManagement/MessageToWorkItemMapper.cs

```diff
@@ -48,7 +53,12 @@ public class MessageToWorkItemMapper
                 return appendOnlyId.Value;
             }
 
-            // Just a standard conversation - look up the cache based on the conversation ID
+            // Just a standard conversation - look up the cache based on the conversation ID (or guid)
+            if (_useConversationGuid)
+            {
+                return GetWorkItemIdFromConversationId(message.ConversationGuid, _workItemsCache);
+            }
+
```

Comment:
I am thinking if this should be more like:
int? workItemId = GetWorkItemIdFromConversationId(message.ConversationGuid, _workItemsCache); 
if (workItemId == null)
{workItemId = GetWorkItemIdFromConversationId(message.ConversationIndex, _workItemsCache);}
return workItemId;

Then in GetWorkItemIdFromConversationId() you don't need the second part.